README.md

```diff
@@ -1,134 +1,121 @@
+3. Configure the `Settings` for the select. The `internalMsg` field takes a function that expects a `Select.Msg` and returns the `Msg` we defined in step 2. Please refer to the module documentation for more information on the settings.
 
 ```elm
-type alias Settings msg a =
-    { internalMsg : ( Msg a, Maybe a ) -> msg
-    , optionType : String
-    , optionLabel : a -> String
-    , optionDescription : a -> String
-    , searchFn : SearchUnion a
-    , debounceDuration : Float
-    , characterSearchThreshold : Int
-    , closeOnSelect : Bool
+selectSettings : Select.Settings Msg Product
+selectSettings =
+    { internalMsg = \msg -> HandleSelectUpdate msg
+    , searchFn =
+        \searchText products ->
+            List.filter (\product -> String.contains (String.toLower searchText) (String.toLower product.name)) products
+    , optionType = "Product"
+    , optionLabel = \product -> product.name
+    , optionDescription = \product -> product.price
+    , optionsContainerMaxHeight = 300
+    , optionHeight = 50
+    , closeOnSelect = True
     }
 ```
 
-- The `internalMsg` field takes a function that expects a tuple containing a SmartSelect.Msg and the selection state and returns the msg we defined in step 2.
-- `optionType` is a string that indicates what kind of data is being selected, i.e. "Product" or "Client"
-- `optionLabel` expects an instance of the data being selected from and returns a string naming/labeling the instance, i.e. if it is a "Product" being selected, the label may be "Garden Hose"
-- `optionDescription` expects an instance of the data being selected from and returns a string describing the instance, i.e. if the label is "Garden Hose", the description may be "30 ft"
-    - Because the smart select is unaware of the type and structure of the data it is processing, these functions are necessary to help render the options in the select dropdown.
-- `debounceDuration` indicates how long if at all to wait between the last keypress and executing a search. This is particularly useful if the search being executed is pinging an external source.
-- `characterThreshold` indicates how many if any characters should be typed before a search is executed.
-- `closeOnSelect` indicates whether or not the `SmartSelect` should close itself after a selection has been made.
-
-The `searchFn` field expects a `SearchUnion`. The `SearchUnion` type as defined in the `SmartSelect.Utilities` module is below:
+4. Initialize the select. The `init` function takes in the select settings and a previously selected element, if any. In the case of the `Single` pickers, the selected element is a `Maybe a` whereas in the `Multi` pickers selected is a `List a`.
 
 ```elm
-type SearchUnion a
-    = Local (String -> List a)
-    | API (Color.Color, ApiSearchAttrs a)
-```
+init : ( Model, Cmd Msg )
+init =
+    ( { products = products
+      , select = Select.init selectSettings Nothing
```

Comment:
I don't like the Nothing here. Everyone will be wondering what it is. Maybe we can move it into the settings as `initialSelection` or create our own type that says `NothingSelected`.

Reply:
I feel like the documentation is pretty clear on what that second argument is. We could do the type for added clarity, but we definitely need the argument to initialize the selection in the event a selection has been previously made.

Reply:
Why can't you just move it into the settings?

Reply:
I suppose we could... then to init with a preselection we'd do something like the following: (?)

```elm
init : ( Model, Cmd Msg )
init =
    ( { ...
      , select = Select.init selectSettings |> SmartSelect.setSelected previouslySelected
    } )
```

How does that look?

Reply:
I like that. It would also allow you to set the selection later.